crates/bevy_asset/src/asset_changed.rs

```diff
@@ -0,0 +1,475 @@
+//! Filter query to limit iteration over [`Handle`]s which content was updated recently.
+use std::marker::PhantomData;
+
+use bevy_ecs::{
+    archetype::{Archetype, ArchetypeComponentId},
+    component::{ComponentId, Tick},
+    prelude::{Changed, Entity, Or, Resource, World},
+    query::{Access, FilteredAccess, QueryItem, ReadFetch, ReadOnlyWorldQuery, WorldQuery},
+    storage::{Table, TableRow},
+    world::unsafe_world_cell::UnsafeWorldCell,
+};
+use bevy_utils::{get_short_name, HashMap};
+
+use crate::{Asset, AssetId, Handle};
+
+#[doc(hidden)]
```

Comment:
Why hidden docs? This is a public type.

Reply:
I don't want this to be part of the public API. It has to be public because the `asset_events` system accesses it, and is itself public. But otherwise it shouldn't be touched by 3rd parties. Bereft the choice of making the type `pub(crate)`, the second best option is making it `#[doc(hidden)]`.

We could probably make `asset_events` private though. What do you think?

Reply:
My two cents rule for my own projects is:
- label all systems so users can order theirs relative to mine
- make systems themselves non public

That works around the issue here I think. But not sure this is the best approach.

Otherwise I'm fine with the explanation. Just wanted to confirm this was intended. Maybe leave a comment to explain?

Reply:
That's generally my preferred approach as well.